### ISSUE TITLE
On alacritty some keys are being repeated

### ISSUE THREAD

**Issue by listout** [neovim/neovim#31806]
### Problem

After commit 4fb3b57a19cb83a99d4c5489982c22d96e28e8c2 some keys (for example Tab and Backspace) are being repeated. For example in normal mode, when deleting work using backspace two alphabets are being deleted.

### Steps to reproduce

build vim from master (after 4fb3b57a19cb83a99d4c5489982c22d96e28e8c2)
Type in some sentences, insert normal mode, try deleting using backspace.

### Expected behavior

Keys are not repeated (or only one alphabet is removed deleted when using backspace)

As it used to be before commit 4fb3b57a19cb83a99d4c5489982c22d96e28e8c2 was merged.

### Nvim version (nvim -v)

NVIM v0.11.0-dev-1444+g4fb3b57a19

### Vim (not Nvim) behaves the same?

No

### Operating system/version

Linux

### Terminal name/version

alacritty 0.14.0

### $TERM environment variable

xterm-256color

### Installation

From git

**Comment by gpanders** [comment]
Alacritty is likely sending repeat events for Enter, Tab, and Backspace, which they should not be doing in accordance with the spec https://sw.kovidgoyal.net/kitty/keyboard-protocol/#report-event-types

You can confirm this by installing `kitty` and, in Alacritty, running `printf 'x1b[>3u' ; kitty +kitten show_key`. When you press Enter, Backspace, or Tab you _should_ see only one key press for each, not two (you'll have to force kill this command or close the window since it won't be possible to Ctrl-C out of it).

**Comment by listout** [comment]
I can confirm; on kitty and gnome terminal this does not happen.

**Comment by gpanders** [comment]
If you file an issue with Alacritty, please link it here so other users can track it. I'm going to close this one since there's nothing actionable on Nvim's side.

**Comment by gpanders** [comment]
In the meantime, you can disable key event reporting in Neovim by applying the following patch:

```diff
diff --git a/src/nvim/tui/tui.c b/src/nvim/tui/tui.c
index f4337d5011..5f261a4aed 100644
--- a/src/nvim/tui/tui.c
+++ b/src/nvim/tui/tui.c
@@ -299,7 +299,7 @@ void tui_set_key_encoding(TUIData *tui)
     // Progressive enhancement flags:
     //   0b01   (1) Disambiguate escape codes
     //   0b10   (2) Report event types
-    out(tui, S_LEN("\x1b[>3u"));
+    out(tui, S_LEN("\x1b[>1u"));
     break;
   case kKeyEncodingXterm:
     out(tui, S_LEN("\x1b[>4;2m"));
```

**Comment by listout** [comment]
https://github.com/alacritty/alacritty/issues/8385

**Comment by listout** [comment]
So the issue was reproducible with foot terminal too

**Comment by gpanders** [comment]
foot has the same bug https://codeberg.org/dnkl/foot/issues/1892

**Comment by timmyha** [comment]
I am experiencing this in both kitty and foot. I just updated from nvim 9.5 to 10 (built from source) when the issue started.

**Comment by clason** [comment]
Yes, that version is too old, as the discussion in the linked alacritty issue should make clear. You need to update.

**Comment by bluz71** [comment]
The version of Alacritty and Foot in the Ubuntu LTS release are currently `0.13.2` and `1.16.2-2ubuntu0.1` respectively (old versions with broken duplicate key behaviour).

Releasing Neovim `0.11` as is will break a lot people's working setup in a very weird way. 

Running nightly for the first time today (normally `0.10.X` user) resulted in me going down a rabbit hole of _"what the heck is going on, I can't backspace now without multiple characters being deleted???"_.

Also note, Alacritty does **not** release a Debian or Linux binary package ready for easy installation. So just telling folks to upgrade Alacritty on Linux is not as easy as it sounds. Ubuntu snaps are not viable either (at least for me) since Linux Mint (Ubuntu derivative) disables Snaps.

Is there no way at all to provide compatibility for folks running older versions of Alacritty and Foot provided by their Linux distribution of choice? If not, then maybe a startup warning message or `:checkhealth` message?

This will cause grief if released as is.

**Comment by gpanders** [comment]
We will temporarily disable key event reporting before the 0.11 release and then re-enable it on nightly again afterwards.

**Comment by bluz71** [comment]
Quite reasonable indeed, thanks @gpanders, a very good strategy (for near term).

**Comment by daveyarwood** [comment]
I just started having this issue on Ubuntu 22.04, Kitty 0.31.0, Neovim 0.11.0-dev.

I saw the suggestion above to update Kitty to the latest version (currently 0.39.1), so I did that, and that fixed the issue.

**Comment by fedreg-bn** [comment]
after seeing the linked PR for alacritty, https://github.com/alacritty/alacritty/issues/8385,  I assumed updating to latest alacritty (15.1) would resolve..  But still seeing the same duplicate keypress behavior on latest nvim 
```
NVIM v0.11.0-dev
Build type: RelWithDebInfo
LuaJIT 2.1.0-beta3
```
Are folks still seeing this issue?

**Comment by listout** [comment]
> Are folks still seeing this issue?

I'm on alacritty 0.15.1 and I don't see the issue anymore.

**Comment by Chainso** [comment]
@fedreg-bn I'm also getting this after updating alacritty for alacritty `v0.15.1` and `0.16.0-dev` with neovim `v0.11.0-dev`

**Comment by fedreg-bn** [comment]
@listout thanks!  Are you also running nvim `v0.11.0-dev`?

**Comment by listout** [comment]
> [@listout](https://github.com/listout) thanks! Are you also running nvim `v0.11.0-dev`?

Yes

```
NVIM v0.11.0-dev-1785+gb0bbe25c48
Build type: RelWithDebInfo
LuaJIT 2.1.1736781742
```

**Comment by SaschaMendel** [comment]
Still experiencing this issue with following versions:

NVIM v0.11.0-dev
Build type: RelWithDebInfo
LuaJIT 2.1.1703358377

alacritty 0.15.1

**Comment by mr-alimuhammadi** [comment]
I had the same problem: Enter, Backspace, and Tab keys were repeating unexpectedly only in Neovim, and only when Alacritty was set as the default terminal (e.g., via update-alternatives).

Turns out it was an environment issue — Alacritty wasn’t launching Neovim with a compatible $TERM, which caused the weird input behavior.

Solution: Create a wrapper for Alacritty that sets the correct $TERM

Create a script at ~/.local/bin/alacritty-wrapper:
```bach
#!/bin/bash
env TERM=xterm-256color alacritty "$@"
```
Make it executable:
```bash
chmod +x ~/.local/bin/alacritty-wrapper
```

Set it as the default terminal:
```bash
sudo update-alternatives --install /usr/bin/x-terminal-emulator x-terminal-emulator ~/.local/bin/alacritty-wrapper 100
sudo update-alternatives --config x-terminal-emulator
```

**Comment by dror-g** [comment]
> env TERM=xterm-256color alacritty "$@"

that's not it for me. `TERM` is already set to `xterm-256color`.
What solved it was updating Alacritty to 0.15.1 ,
with Nvim 0.12.0-dev.

**Comment by krysopath** [comment]
I have the same issue on alacritty 0.13.2 and kitty 0.32.2, not on older terminal emulators.

However I reproduced this with the version in 

- ubuntu lts apt repo
- ppa nightly build v0.12


but not with self compiled v0.11.6, just yesterday

the self compiled v0.11.6 does NOT exhibit this behaviour on kitty and lacritty.
